receiver/hostmetricsreceiver/internal/scraper/cpuscraper/metadata.yaml

```diff
@@ -18,3 +18,11 @@ metrics:
       aggregation: cumulative
       monotonic: true
     attributes: [cpu, state]
+
+  system.cpu.utilization:
+    enabled: true
```

Comment:
Hey @rubenruizdegauna @jlegoff. We are currently working on the metrics builder that right now introduces a way to disable/enable metrics, but going forward it'll provide a way to convert metrics from cumulative to delta in the user config. Here is the tracking issue https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/6221

Will that suit your needs? If not, and these utilization metrics are really needed, please make them optional `enabled: false`. Otherwise we will be sending essentially the same data in two formats by default which is not desirable.

Reply:
Hi @dmitryax, is this the correct issue? You pointed to the one I created with the missing metrics.

Would the metric builder behave like the cumulativetodelta processor? I'm not sure that's what we need in this case: here, we want to divide the delta by the time between measurement, as defined in the [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/README.md#instrument-naming):

> system.cpu.utilization is defined as the difference in system.cpu.time measurements divided by the elapsed time.

Do the upcoming changes in the metric builder contemplate this?

I don't have a strong optinion about making utilization metrics optional, as long as they are available :) But keep in mind that these metrics are all listed in the [semantic conventions](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/system-metrics.md) for system metrics.

Reply:
Sorry, here is the correct issue: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/6429

> Would the metric builder behave like the cumulativetodelta processor? 

yes, so it'll be a bit different representation except if scrape interval is 1 second.

If there are no other objections, we can probably keep both metrics as long as one of them is disable to avoid data duplication by default.

Reply:
Hi @dmitryax, I think that it is a good idea to keep both as they provide different values, and as @jlegoff mentions, the one added in the PR follows the utilization spec.

I tried `make generate` in my environment with the metric `enabled: false` and it fails claiming that `enabled` is mandatory, as if the option wouldn't be there. I checked the mdatagen tests, and it seems that it only accepts `enabled: true` values.

For instance, the test for `no_enabled.yaml` passes even when the option is there but with `false` value.

```
name: metricreceiver
metrics:
  system.cpu.time:
    enabled: false
    description: Total CPU seconds broken down by different states.
    unit: s
    sum:
      value_type: double
      monotonic: true
      aggregation: cumulative
    attributes:

```

Am I missing something? Is this the correct way to have it disabled by default?

Reply:
> Hi @dmitryax, I think that it is a good idea to keep both as they provide different values, and as @jlegoff mentions, the one added in the PR follows the utilization spec.

They do provide different values but can be translated between each other, e.g "time cumulative -> time delta divided by scrape interval" gives utilization. So we shouldn't send them both by default.

And since `system.cpu.time` is being used for a while already, I think we should disable `system.cpu.utilization` by default unless any of @open-telemetry/collector-contrib-approvers has different opinion.

> I tried make generate in my environment with the metric enabled: false and it fails claiming that enabled is mandatory, as if the option wouldn't be there. I checked the mdatagen tests, and it seems that it only accepts enabled: true values.

Oh this seems like a bug. We didn't catch it because it's new feature and we did'n have optional metrics up to this point. Let me submit a fix for that.

Reply:
> They do provide different values but can be translated between each other, e.g "time cumulative -> time delta divided by scrape interval" gives utilization. So we shouldn't send them both by default.
> 
> And since system.cpu.time is being used for a while already, I think we should disable system.cpu.utilization by default unless any of @open-telemetry/collector-contrib-approvers has different opinion. And I think Cumulative Sum works better for further aggregations/rollups than Gauge.

👍  It sounds reasonable

> Oh this seems like a bug. We didn't catch it because it's a new feature and we didn't have optional metrics up to this point. Let me submit a fix for that.

Great! Thanks!!

Reply:
> I think Cumulative Sum works better for further aggregations/rollups than Gauge.

I think gauge is actually what we want here. The spec suggests [system.cpu.utilization](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/system-metrics.md#systemcpu---processor-metrics) should be a gauge. I think this is true for all utilization metrics since it is not generally meaningful to sum a ratio over time.

Reply:
I'm not saying that `system.cpu.utilization` shouldn't be Gauge, I mean `system.cpu.time` is Cumulative Sum which is better for aggregation/rolloups. So just another argument to keep only `system.cpu.time` enabled by default

Reply:
Submitted a PR to fix the bug not allowing `enabled: false` https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/7166